core/chains/evm/txmgr/common.go

```diff
@@ -90,3 +95,23 @@ func stringToGethHash(s string) (h common.Hash, err error) {
 	err = h.UnmarshalText([]byte(s))
 	return
 }
+
+func ToGethFees[FEE txmgrtypes.Fee](f FEE) (fee gas.EvmFee, err error) {
+	b, err := json.Marshal(f)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(b, &fee)
+	return
+}
+
+func ToGenericReceipt[TX_HASH types.Hashable, R txmgrtypes.ChainReceipt[TX_HASH]](r *evmtypes.Receipt) (receipt R, err error) {
+	b, err := json.Marshal(r)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(b, &receipt)
+	return
+}
```

Comment:
This is unusual. Is it a workaround? Seems worth a comment at least.

Reply:
yes, these are just work arounds - the chain-client interfaces need to use generic types before these can be removed. right now they are using the evm client directly.

I will add some comments to these functions